go/mysql/query.go

```diff
@@ -303,7 +303,11 @@ func (c *Conn) parseRow(data []byte, fields []*querypb.Field, reader func([]byte
 //	 2. if the server closes the connection when a command is in flight,
 //	    readComQueryResponse will fail, and we'll return CRServerLost(2013).
 func (c *Conn) ExecuteFetch(query string, maxrows int, wantfields bool) (result *sqltypes.Result, err error) {
-	result, _, err = c.ExecuteFetchMulti(query, maxrows, wantfields)
+	var more bool
+	result, more, err = c.ExecuteFetchMulti(query, maxrows, wantfields)
+	if more {
+		return nil, vterrors.Errorf(vtrpc.Code_INTERNAL, "unexpected multiple results. Use ExecuteFetchMulti instead")
+	}
```

Comment:
Question: should we iterate and consume all results?

Reply:
I think we must here, otherwise we leave the connection in an invalid state. And a subsequent query would see the previous result here.

Reply:
Done.

Reply:
@shlomi-noach are there other places in the code base where we pass in 0 incorrectly and do want to consume all the results?

Reply:
There's not other explicit ` c.ReadQueryResult(0, ...)` call in the code, FWIW.

Reply:
@harshit-gangal further edited the `ExecuteFetch`/drain logic to fix potential leaks, and consolidated the draining logic. I think we should be good now.